rasa/core/story_conflict.py

```diff
@@ -0,0 +1,252 @@
+
+            if not prev_event_type and k.startswith(MESSAGE_INTENT_ATTRIBUTE + "_"):
+                prev_event_type = "intent"
+                prev_event_name = k[len(MESSAGE_INTENT_ATTRIBUTE + "_") :]
+
+        return prev_event_type, prev_event_name
+
+    def add_conflicting_action(self, action: Text, story_name: Text):
+        """
+        Add another action that follows from the same state
+        :param action: Name of the action
+        :param story_name: Name of the story where this action
+        is chosen
+        """
+        if action not in self._conflicting_actions:
+            self._conflicting_actions[action] = [story_name]
+        else:
+            self._conflicting_actions[action] += [story_name]
+
+    @property
+    def conflicting_actions(self) -> List[Text]:
+        """
+        Returns the list of conflicting actions
+        """
+        return list(self._conflicting_actions.keys())
+
+    @property
+    def conflicting_actions_with_counts(self) -> List[Text]:
+        """
+        Returns a list of strings, describing what action
+        occurs how often
+        """
+        return [f"{a} [{len(s)}x]" for (a, s) in self._conflicting_actions.items()]
+
+    @property
+    def incorrect_stories(self) -> List[Text]:
+        """
+        Returns a list of story names that have not yet been
+        corrected.
+        """
+        if not self.correct_response:
+            # Return all stories
+            return [v[0] for v in self._conflicting_actions.values()]
+
+        incorrect_stories = []
+        story_lists_with_uncorrected_responses = [
+            s
+            for (a, s) in self._conflicting_actions.items()
+            if a != self.correct_response
+        ]
+        for stories in story_lists_with_uncorrected_responses:
+            for story in stories:
+                incorrect_stories.append(story)
+        return incorrect_stories
+
+    @property
+    def has_prior_events(self) -> bool:
+        """
+        Returns True iff anything has happened before this
+        conflict.
+        """
+        return self._get_prev_event(self.sliced_states[-1])[0] is not None
+
+    def story_prior_to_conflict(self) -> Text:
+        """
+        Generates a story string, describing the events that
+        lead up to the conflict.
+        """
+        result = ""
+        for state in self.sliced_states:
+            if not state:
+                continue
+
+            event_type, event_name = self._get_prev_event(state)
+            if event_type == "intent":
+                result += f"* {event_name}\n"
+            else:
+                result += f"  - {event_name}\n"
+
+        return result
+
+    def __str__(self):
+        # Describe where the conflict occurs in the stories
+        last_event_type, last_event_name = self._get_prev_event(self.sliced_states[-1])
+        if last_event_type:
+            conflict_string = f"CONFLICT after {last_event_type} '{last_event_name}':\n"
```

Comment:
suggestion: don't use hungarian notation (https://en.wikipedia.org/wiki/Hungarian_notation)
```suggestion
            conflict_message = f"CONFLICT after {last_event_type} '{last_event_name}':\n"
```